nessie/src/main/java/org/apache/iceberg/nessie/NessieViewOperations.java

```diff
@@ -0,0 +1,153 @@
+  private final Map<String, String> catalogOptions;
+  private IcebergView icebergView;
+
+  NessieViewOperations(
+      ContentKey key,
+      NessieIcebergClient client,
+      FileIO fileIO,
+      Map<String, String> catalogOptions) {
+    this.key = key;
+    this.client = client;
+    this.fileIO = fileIO;
+    this.catalogOptions = catalogOptions;
+  }
+
+  @Override
+  public void doRefresh() {
+    try {
+      client.refresh();
+    } catch (NessieNotFoundException e) {
+      throw new RuntimeException(
+          String.format(
+              "Failed to refresh as ref '%s' is no longer valid.", client.getRef().getName()),
+          e);
+    }
+    String metadataLocation = null;
+    Reference reference = client.getRef().getReference();
+    try {
+      Content content = client.getApi().getContent().key(key).reference(reference).get().get(key);
+      LOG.debug("Content '{}' at '{}': {}", key, reference, content);
+      if (content == null) {
+        if (currentMetadataLocation() != null) {
+          throw new NoSuchViewException("View does not exist: %s in %s", key, reference);
+        }
+      } else {
+        this.icebergView =
+            content
+                .unwrap(IcebergView.class)
+                .orElseThrow(
+                    () -> {
+                      if (content instanceof IcebergTable) {
+                        return new AlreadyExistsException(
+                            "Table with same name already exists: %s in %s", key, reference);
+                      } else {
+                        return new IllegalStateException(
+                            String.format(
+                                "Cannot refresh Iceberg view: Nessie points to a non-Iceberg object for path: %s.",
+                                key));
+                      }
+                    });
+        metadataLocation = icebergView.getMetadataLocation();
+      }
+    } catch (NessieNotFoundException ex) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchViewException("View does not exist: %s in %s", key, reference);
+      }
+    }
+    refreshFromMetadataLocation(metadataLocation, null, 2, l -> loadViewMetadata(l, reference));
+  }
+
+  private ViewMetadata loadViewMetadata(String metadataLocation, Reference reference) {
+    ViewMetadata metadata = ViewMetadataParser.read(io().newInputFile(metadataLocation));
+    Map<String, String> newProperties = Maps.newHashMap(metadata.properties());
+    newProperties.put(NessieTableOperations.NESSIE_COMMIT_ID_PROPERTY, reference.getHash());
+
+    return ViewMetadata.buildFrom(
+            ViewMetadata.buildFrom(metadata).setProperties(newProperties).build())
+        .setMetadataLocation(metadataLocation)
+        .build();
+  }
+
+  @Override
+  public void doCommit(ViewMetadata base, ViewMetadata metadata) {
+    String newMetadataLocation = writeNewMetadataIfRequired(metadata);
+
+    AtomicBoolean failure = new AtomicBoolean(false);
```

Comment:
why does this have to be an `AtomicBoolean`?

Reply:
Because I need to set the failure from `NessieUtil.handleExceptionsForCommits` also. I cannot return a boolean from that method as it throws exception.

Reply:
removed `AtomicBoolean` as it can be simplified.